src/main/java/mate/academy/config/SecurityConfig.java

```diff
@@ -22,4 +32,25 @@ protected void configure(AuthenticationManagerBuilder auth) throws Exception {
         auth.userDetailsService(userDetailsService)
                 .passwordEncoder(passwordEncoder);
     }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .httpBasic().disable()
+                .csrf().disable()
+                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+                .and()
+                .authorizeRequests()
+                .antMatchers("/register", "/login", "/inject")
+                .permitAll()
+                .antMatchers(HttpMethod.GET, "/products", "/users").hasRole(ROLE_USER)
+                .antMatchers(HttpMethod.POST, "/products").hasRole(ROLE_USER)
+                .antMatchers(HttpMethod.DELETE).hasRole(ROLE_ADMIN)
```

Comment:
usually, only users with the ADMIN role have access to the GET /users endpoint. why can users add products?) How about HttpMethod.POST, "/users"?